base-compat-batteries/src/Data/Type/Equality/Compat.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE CPP, NoImplicitPrelude, PackageImports #-}
+#if __GLASGOW_HASKELL__ >= 702
+{-# LANGUAGE Trustworthy #-}
+#endif
+#if __GLASGOW_HASKELL__ >= 708
+{-# LANGUAGE ExplicitNamespaces #-}
+#endif
+module Data.Type.Equality.Compat (
+  -- * The equality types
+  (:~:)(..),
+#if __GLASGOW_HASKELL__ >= 802
+  type (~~),
+#endif
+#if __GLASGOW_HASKELL__ >= 800
+  (:~~:)(..),
+#endif
+
+  -- * Working with equality
+  sym, trans, castWith, gcastWith,
+#if __GLASGOW_HASKELL__ >= 706
+  apply,
+#endif
+#if __GLASGOW_HASKELL__ >= 708
+  inner, outer,
```

Comment:
As I note in https://github.com/phadej/some/pull/2/files/778469cc2f72b44862c2c12c93254e33ab22dd84..c296e33d0a76590f0e6fa6e18c4dadc32d1565ef#diff-533148f9e61bee9cf4d20acc82d92e23, we can actually provide implementations of `inner` and `outer` on more versions of GHC than this.

Reply:
Done.